### ISSUE TITLE
Cross-validation in keras in R: model is inheriting weights from the previous fold

### ISSUE THREAD

**Issue by VictorAlves2100** [rstudio/keras#1149]
Python
Tensorflow: v2.1.0
Keras: v2.3

**R
Tensorflow: v2.3.0

I am performing 4-fold cross-validation acording to the example presenting in the book Deep Learning with R from François Chollet with J. J. Allaire. The model is compiled at each fold. I found that the models are inheriting weights of the previous fold because the performing both in training and in validation sets is improving from the 1 to the 4-fold, and if I if i reverse the training ( from 4 to 1) the reverse occurs. So I conclude that there is an information leakage from the validation set partition to the training partitions and this is occuring because the information stored in the weights.

I tried diferent things no one is working, excepting restart R and clear environment. 

I have put the following withing the loop with no successs:

rm(model) _or_ model<-NULL
K<-backend
K$clear_session() _or_ tf$keras$backend$clear_session()
gb()

**Comment by dfalbel** [comment]
You will usually need to define a new model for each fold.
You can have a function like:

```
define_and_compile_model <- function() {
   model <- keras_model_sequential() %>% 
      ...
   model %>% compile(...)
   model
}
```

And then use it to redefine the model for each fold.
